src/bin/tikv-fail.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![feature(plugin)]
+#![cfg_attr(feature = "dev", plugin(clippy))]
+
+/// Inject failures to `TikV`.
+///
+/// TODO: Integrate into tikv-ctl.
+///
+/// # Examples
+///
+/// ```sh
+/// ./tikv-fail -a 127.0.0.1:9090 inject\
+//      -f tikv::raftstore::store::store::raft_between_save\
```

Comment:
can we support regexp like using "raft_between_save" directly?

Reply:
`tikv::raftstore::store::store::raft_between_save` is too long.

Reply:
Seems can't, unless tikv-fail knows all possible failpoints on remote servers. `ListFailPoints` returns failpoints that are injected already.

Reply:
maybe you can get the fail point list at first, and use a match rule like checking action name at first, then plus structure name, then plus mod name, etc.

In most cases, we will use different action names directly, rarely use duplicated names. And If we support passing injection list in one command, the cost of getting the fail list at first can be neglected.

Reply:
I mean there is no way to get a full list of fail points on a remote server.
`ListFailpoints` returns nothing if there is no injected[1] failpoints, so I can't get the fail point list at first.

[1] injected: it is not something we hard code in tikv source file, it's injected from tikv-fail. eg.
```
// raft.rs
fail_points("a");
fail_points("b");

tikv-fail inject a off => tik-fail list -> a
tikv-fail inject b off => tik-fail list -> a, b
tikv-fail recover a => tik-fail list -> b
```

Reply:
so we have no way to get the registered fail point?

Reply:
Yes